src/components/UI/FilterableChannelSelector.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div>
+    <filter-input v-model="query" on-secondary disable-ime />
+    <form-selector
+      v-model="value"
+      :options="channelOptions"
+      :class="$style.form"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import FilterInput from '/@/components/UI/FilterInput.vue'
+import { useChannelsStore } from '/@/store/entities/channels'
+import useChannelFilter from '/@/components/Main/NavigationBar/NavigationContent/composables/useChannelFilter'
+import FormSelector from '/@/components/UI/FormSelector.vue'
+import { computed, watchEffect } from 'vue'
+import useChannelPath from '/@/composables/useChannelPath'
+import { compareStringInsensitive } from '/@/lib/basic/string'
+import { useModelValueSyncer } from '/@/composables/useModelSyncer'
+
+const props = withDefaults(
+  defineProps<{
+    modelValue?: string | null
+    label?: string
+  }>(),
+  { modelValue: '', onSecondary: false }
+)
+
+const emit = defineEmits<{
+  (e: 'update:modelValue', _val: string | null): void
+}>()
+
+const value = useModelValueSyncer(props, emit)
+
+const { channelsMap } = useChannelsStore()
+
+const channelListForFilter = computed(() =>
+  [...channelsMap.value.values()].filter(channel => !channel.archived)
+)
+const { query, filteredChannels } = useChannelFilter(channelListForFilter)
+const { channelIdToPathString } = useChannelPath()
+
+const channelOptions = computed(() => {
+  return filteredChannels.value
+    .map(channel => ({
+      key: channelIdToPathString(channel.id, true),
+      value: channel.id
+    }))
+    .sort((a, b) => compareStringInsensitive(a.key, b.key))
+})
+
+watchEffect(() => {
+  value.value = (query.value && filteredChannels.value[0]?.id) || value.value
```

Comment:
```suggestion
  if (query.value === '') return

  const firstCandidate = filteredChannels.value[0]
  if (firstCandidate) {
    value.value = firstCandidate.id
  }
```
かなり省略した書き方になってますが、意図としてはこうっぽそうです？ 🧐

個人的には候補が1つでないなら勝手にセットされないほうが嬉しい感じがします

Reply:
意図はそのとおりです。 書き換えた方がいいですかね…？
それと"勝手にセットされない"、って具体的にどういう状況ですか？

Reply:
そのときって代わりに 「~~件ヒット」みたいなのが表示されてる感じですか？

Reply:
想定は元の値が表示されてることだったんですが、今考えると変わらないのも変な気がしますね…
うまい感じのが思いつかないので一旦忘れてください 🙏